src/lib/sql/sql_translator.cpp

```diff
@@ -1117,6 +1132,50 @@ void SQLTranslator::_translate_select_groupby_having(const hsql::SelectStatement
   }
 }
 
+void SQLTranslator::_translate_set_operation(const hsql::SetOperation& set_operator) {
+  const auto& left_input_lqp = _current_lqp;
+
+  // Create a nested SQL-Translator for the right hand site of the Set-Statement to not make changes in the current one
```

Comment:
Could you clarify what type of changes you mean?

Reply:
I mean we don't want any of the member variables to be changed since we are reusing the current SQLTranslator further down the line for more SetOperations in the vector.

Not sure, if this is clear and tbh also not certain if it would be strictly necessary